packages/oft-evm/test/OFT.t.sol

```diff
@@ -373,6 +395,74 @@ contract OFTTest is TestHelperOz5 {
         assertEq(cERC20Mock.balanceOf(address(cOFTAdapter)), 0);
     }
 
+    function test_native_oft_adapter_debit() public virtual {
+        uint256 amountToSendLD = 1 ether;
+        uint256 minAmountToCreditLD = 1 ether;
+        uint32 dstEid = D_EID;
+
+        vm.prank(userD);
+        vm.expectRevert(
+            abi.encodeWithSelector(IOFT.SlippageExceeded.selector, amountToSendLD, minAmountToCreditLD + 1)
+        );
+        dNativeOFTAdapter.asNativeOFTAdapterMock().debit(amountToSendLD, minAmountToCreditLD + 1, dstEid);
+
+        vm.prank(userD);
+        (uint256 amountDebitedLD, uint256 amountToCreditLD) = dNativeOFTAdapter.asNativeOFTAdapterMock().debit(
+            amountToSendLD,
+            minAmountToCreditLD,
+            dstEid
+        );
+
+        assertEq(amountDebitedLD, amountToSendLD);
+        assertEq(amountToCreditLD, amountToSendLD);
+    }
+
+    function test_native_oft_adapter_credit() public {
+        uint256 amountToCreditLD = 1 ether;
+        uint32 srcEid = D_EID;
+
+        // simulate userD already having deposited native to the adapter
+        vm.deal(address(dNativeOFTAdapter), amountToCreditLD);
+
+        uint256 amountReceived = dNativeOFTAdapter.asNativeOFTAdapterMock().credit(userB, amountToCreditLD, srcEid);
+
+        assertEq(userB.balance, initialNativeBalance + amountReceived);
+        assertEq(address(dNativeOFTAdapter).balance, 0);
+    }
+
+    function test_native_oft_adapter_send() public virtual {
+        assertEq(userD.balance, initialNativeBalance);
+        assertEq(address(dNativeOFTAdapter).balance, 0);
+
+        uint256 amountToSendLD = 1 ether;
+        uint32 dstEid = B_EID;
+
+        SendParam memory sendParam = SendParam(
+            dstEid,
+            addressToBytes32(userB),
+            amountToSendLD,
+            amountToSendLD,
+            OptionsBuilder.newOptions().addExecutorLzReceiveOption(200000, 0),
+            "",
+            ""
+        );
+
+        MessagingFee memory fee = dNativeOFTAdapter.quoteSend(sendParam, false);
+        uint256 msgValue = fee.nativeFee + sendParam.amountLD;
+
+        vm.prank(userD);
+        vm.expectRevert(
+            abi.encodeWithSelector(NativeOFTAdapter.InsufficientMessageValue.selector, fee.nativeFee, msgValue)
+        );
+        dNativeOFTAdapter.asNativeOFTAdapterMock().send{ value: fee.nativeFee}(sendParam, fee, userD);
```

Comment:
Two things:

- I would add comments about what's being tested and why it should fail
- I would add the missing case for when the `msgValue` is higher than `fee.nativeFee + sendParam.amountLD` because that triggers a revert as well

Reply:
> I would add comments about what's being tested and why it should fail

Will do!

> I would add the missing case for when the msgValue is higher than fee.nativeFee + sendParam.amountLD because that triggers a revert as well

Oo yes great catch! I'll write this in the next commit

Reply:
I decided to make a separate PR for this to keep things cleaner since this PR has already been sent for audit as is

https://github.com/LayerZero-Labs/devtools/pull/880